### PR TITLE
Added new cell kind DateCell, displaying calendar dates (without time…

### PR DESCRIPTION
**Pull request by hermann-p**
… or locale).

### CODE REVIEW

File: src/main/js/components/cells/date/DateCell.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      if (savedDateString !== this.momentToString(inputDate)) {
+        ActionCreator.changeCell(cell, inputDate.format(DateFormats.formatForServer));
+      }
+    } else { // no saving => reset display value
+      this.setState({value: this.getSavedMoment()})
+    }
+  };
+
+  handleDateUpdate = moment => {
+    console.log("DateCell.handleDateUpdate", moment);
```
Comment (Narigo) [inline src/main/js/components/cells/date/DateCell.jsx]: Forgot to remove?

File: src/main/js/components/cells/date/DateEditCell.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  handleClickClearDate = event => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.props.handleEditFinished(false);
+    ActionCreator.toggleCellEditing(false);
+  };
+
+  handleChange = moment => {
+    if (moment) {
+      this.setState({moment: moment});
```
Comment (Narigo) [inline src/main/js/components/cells/date/DateEditCell.jsx]: `this.setState({moment});` works here

File: src/main/js/constants/TableauxConstants.js [hunk 3]
```diff
@@ -117,6 +118,11 @@ var TableauxConstant = {
     formatForUser: "DD.MM.YYYY - HH:mm"
   },
 
+  DateFormats: {
+    formatForServer: "YYYY-MM-DD",
+    formatForUser: "DD.MM.YYYY"
```
Comment (Narigo) [inline src/main/js/constants/TableauxConstants.js]: Can `formatForUser` be localized somehow?
This is just a question and should be tracked, if we want / need that. American users may want to have MM/DD/YYYY or something...

File: src/main/js/helpers/RowConcatHelper.js [hunk 4]
```diff
@@ -105,6 +105,17 @@ var internal = {
           }
           break;
 
+        case ColumnKinds.date:
+          const dateValue = getCellValueFromLanguage(concatValue, concatColumn);
+          if (!_.isEmpty(dateValue)) {
+            const moment = Moment(dateValue, TableauxConstants.DateFormats.formatForServer);
+            const formattedDateValue = moment.isValid()
+              ? moment.format(TableauxConstants.DateFormats.formatForUser)
+              : ""
```
Comment (Narigo) [inline src/main/js/helpers/RowConcatHelper.js]: `;`

File: src/main/js/components/cells/date/DateEditCell.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+DateEditCell.propTypes = {
+  setCellKeyboardShortcuts: React.PropTypes.func.isRequired,
+  handleDateUpdate: React.PropTypes.func.isRequired,
+  toDisplayValue: React.PropTypes.func.isRequired,
+  handleEditFinished: React.PropTypes.func.isRequired,
+  cell: React.PropTypes.object.isRequired,
+  value: React.PropTypes.object.isRequired,
+  langtag: React.PropTypes.string,
```
Comment (Narigo) [inline src/main/js/components/cells/date/DateEditCell.jsx]: `,` too much

File: src/main/js/components/cells/date/DateCell.jsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  finishedEditing = save => {
+    if (save) {
+      const savedDateString = this.momentToString(this.getSavedMoment());
+      const inputDate = this.state.value;
+      const {cell} = this.props;
+      if (savedDateString !== this.momentToString(inputDate)) {
+        ActionCreator.changeCell(cell, inputDate.format(DateFormats.formatForServer));
+      }
+    } else { // no saving => reset display value
+      this.setState({value: this.getSavedMoment()})
```
Comment (Narigo) [inline src/main/js/components/cells/date/DateCell.jsx]: `;`

File: src/main/js/components/cells/date/DateEditCell.jsx (hunk 2 above)
Comment (hermann-p) [inline src/main/js/components/cells/date/DateEditCell.jsx]: Nice to know...

File: src/main/js/constants/TableauxConstants.js (hunk 3 above)
Comment (hermann-p) [inline src/main/js/constants/TableauxConstants.js]: I agree, we should keep that one in mind.

Comment (alexvetter) [inline src/main/js/constants/TableauxConstants.js]: There is already a Task for "localization". So, it's already tracked. 👍

File: src/main/js/components/cells/date/DateCell.jsx [hunk 7]
```diff
@@ -7,8 +7,15 @@ import DateEditCell from "./DateEditCell";
 import TableauxConstants from "../../../constants/TableauxConstants";
 import Moment from "moment";
 import ActionCreator from "../../../actions/ActionCreator";
+import keyMirror from "keymirror";
 const DateFormats = TableauxConstants.DateFormats;
 
+const OPTIONS = keyMirror({
```
Comment (Narigo) [inline src/main/js/components/cells/date/DateCell.jsx]: At some point, we should check out something different than `keyMirror`. Something that takes a list and generates a map from it simply.

```
const OPTIONS = enum(["SAVE", "CANCEL", "CLEAR"]);
// would generate:
const OPTIONS = {
  "SAVE": "SAVE",
  "CANCEL": "CANCEL",
  "CLEAR": "CLEAR"
};
```

**Pull request merged**